wp-content/mu-plugins/register-rest-routes.php

```diff
@@ -288,3 +290,187 @@
     }
   ));
 });
+
+add_action('rest_api_init', function() {
+  $v = 'api/v1'; // namespace for the current version of the API
+
+  $exp = WEEK_IN_SECONDS; // expiration of the transient caches
+  
+
+  register_rest_route('api/v1', '/confirmation-events/', array(
+    'methods' => 'POST',
+    'permission_callback' => '__return_true',
+
+    'callback' => function(WP_REST_Request $request) {
+
+      $path = ABSPATH . 'wp-content/mu-plugins/config/config.yml';
+
+      $sendgrid_api_key;
+      $sendgrid_confirmation_template_id;
+      $sendgrid_confirmed_template_id;
+      $sendgrid_contact_list_id;
+     
+      //Fetch the Sendgrid Configuration values from config file
+       if (file_exists($path)) {
+         $sendgrid_api = 'SENDGRID_API_KEY';
+         $sendgrid_template_id1 = 'CONFIRMATION_TEMPLATE_ID';
+         $sendgrid_template_id2 = 'CONFIRMED_TEMPLATE_ID';
+         $sendgrid_list_id = 'LIST_ID';
+         $config_val = Spyc::YAMLLoad($path);
+         $sendgrid_values = $config_val['sendgrid'];
+         $sendgrid_api_key = $sendgrid_values[$sendgrid_api];
+         $sendgrid_confirmation_template_id = $sendgrid_values[$sendgrid_template_id1];
+         $sendgrid_confirmed_template_id = $sendgrid_values[$sendgrid_template_id2];
+         $sendgrid_contact_list_id = $sendgrid_values[$sendgrid_list_id];
+       }
+
+      $input = $request->get_body();
+      $data = json_decode($input, true);
+
+      $response = new WP_REST_Response($data);
+
+      if (is_array($data)) {
+        $log_dir = WP_CONTENT_DIR . '/sendgrid-events';
+        $log_file = $log_dir . '/events.log';
+        $logMessage = print_r($data, true);        
+        foreach ($data as $event) {
```

Comment:
- [ ] Create a helper method for logging events, and don't log the user's email address